src/main/java/com/redhat/pantheon/servlet/ModuleRenderServlet.java

```diff
@@ -0,0 +1,128 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.model.module.Metadata;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.model.module.Content;
+import com.redhat.pantheon.model.module.ModuleRevision;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+
+import org.jetbrains.annotations.NotNull;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.conf.GlobalConfig.CONTENT_TYPE;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+
+/**
+ * Get operation to render a Module data in JSON format.
+ * Only two parameters are expected in the Get request:
+ * 1. locale - Optional; indicates the locale that the module content is in, defaulted to en-US
+ * 2. module_id - indicates the uuid string which uniquely identifies a module
+ *
+ * The url to GET a request from the server is /api/module/render
+ * Example: <server_url>/api/module/render?locale=en-us&module_id=xyz
+ * The said url is accessible outside of the system without any authentication.
+ *
+ * @author Ankit Gadgil
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.resourceTypes=sling/servlet/default",
+                "sling.servlet.selectors=query",
+                "sling.servlet.extensions=json",
+                Constants.SERVICE_DESCRIPTION + "=Servlet to facilitate GET operation which accepts locale and module uuid to output module data",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/render")
+public class ModuleRenderServlet extends AbstractJsonSingleQueryServlet {
+    private final Logger log = LoggerFactory.getLogger(ModuleRenderServlet.class);
+
+    @Override
+    protected String getQuery(SlingHttpServletRequest request) {
+        // Get the query parameter(s)
+        String localeParam = ServletUtils.paramValue(request, "locale", DEFAULT_MODULE_LOCALE.toString());
+        String uuidParam = paramValue(request, "module_id", "");
+        String nodeType = paramValue(request, "nodeType","pant:module");
+
+        StringBuilder query = new StringBuilder("select * from ")
+                .append("[").append(nodeType).append("]")
+                .append(" as module WHERE ")
+//                .append("where (isdescendantnode(module, '/content/repositories') ")
+//                .append("or isdescendantnode(module, '/content/modules') ")
+//                .append("or isdescendantnode(module, '/content/sandbox')) ")
```

Comment:
Lets get rid of commented-out code, unless strictly necessary.

Reply:
sure let's remove it, I left it there because we wanted to keep the code for future reference.